util/argo/diff/ignore_test.go

```diff
@@ -187,5 +191,32 @@ func TestIgnoreDiffConfig_HasIgnoreDifference(t *testing.T) {
 		assert.False(t, ok)
 		require.Nil(t, actual)
 	})
+	t.Run("will merge ignore differences correctly", func(t *testing.T) {
```

Comment:
I don't think I follow this test... I think I see that we're supposed to de-dupe some ignore rules. Can you clarify?

Reply:
de-dupe during merge is what this test-case is validating. This is an additional logic improvement but main issue was that it was completely skipping the Application configs if any override config was present. The main scenario is validated in other test-cases.

Reply:
Improved the test-case description to make it more evident 👍🏻